4.assess_fibrosis_reversal/scripts/0.apply_models.py

```diff
@@ -0,0 +1,259 @@
+    print(f"Shape of the model DataFrame: {model_df.shape}")
+
+
+# ## Extract final model predicted probabilities for each treatment
+
+# In[7]:
+
+
+# Create an empty DataFrame to store the results
+combined_prob_df = pd.DataFrame()
+
+# Loop through each model in the models directory
+for model_path in models_dir.iterdir():
+    model_type = model_path.stem.split("_")[5]  # Get the model type
+    
+    # Process each plate's data from the plate_data_dict
+    for plate_name, info in plate_data_dict.items():
+        print(f"Extracting {model_type} probabilities from {plate_name} data...")
+
+        # Load in model to apply to datasets
+        model = load(model_path)
+
+        # Load in label encoder
+        le = load(encoder_path)
+
+        # Get unique cell types and their corresponding encoded values
+        unique_labels = le.classes_
+        encoded_values = le.transform(unique_labels)
+
+        # Create a dictionary mapping encoded values to original labels
+        label_dict = dict(zip(encoded_values, unique_labels))
+
+        # Load in the DataFrame associated with the plate
+        data_df = info["model_df"].reset_index(drop=True)
+
+        # Load in X data to get predicted probabilities
+        X, _ = get_X_y_data(df=data_df, label="Metadata_cell_type")
+
+        # Predict class probabilities for morphology feature data
+        predicted_probs = model.predict_proba(X)
+
+        # Storing probabilities in a pandas DataFrame
+        prob_df = pd.DataFrame(predicted_probs, columns=model.classes_)
+
+        # Update column names in prob_df using the dictionary and add suffix "_probas"
+        prob_df.columns = [label_dict[col] + '_probas' for col in prob_df.columns]
+
+        # Add a new column called predicted_label for each row
+        prob_df['predicted_label'] = prob_df.apply(lambda row: row.idxmax()[:-7], axis=1)
+
+        # Select metadata columns from the data
+        metadata_columns = data_df.filter(like='Metadata_')
+
+        # Combine metadata columns with predicted probabilities DataFrame based on index
+        prob_df = prob_df.join(metadata_columns)
+
+        # Add a new column for model_type
+        prob_df['model_type'] = model_type
+        
+        # Append the probability DataFrame to the combined DataFrame
+        combined_prob_df = pd.concat([combined_prob_df, prob_df], ignore_index=True)
+
+# Save combined prob data
+combined_prob_df.to_csv(f"{prob_dir}/combined_batch_1_predicted_proba.csv", index=False)
+
+
+# In[8]:
+
+
+# Filter rows where Metadata_treatment is 'DMSO'
+dmso_rows = combined_prob_df[combined_prob_df['Metadata_treatment'] == 'DMSO']
+
+# Select only the desired columns
+selected_columns = ['Failing_probas', 'Healthy_probas', 'predicted_label', 'Metadata_cell_type']
+dmso_rows_filtered = dmso_rows[selected_columns]
+
+# Display the filtered DataFrame
+dmso_rows_filtered.head(100)
```

Comment:
Why 100?

Reply:
FWIW I do this too haha

Reply:
Oh shoot, I was doing some testing lol relatable. I will remove it, it doesn't need to be printed anyway haha